src/components/movieForm/movieForm.js

```diff
@@ -89,55 +126,71 @@ export const MovieForm = ({ movie, onSubmit }) => {
             <label className="form-label" htmlFor="rating">
               RATING
             </label>
             <input
               className="form-input form-placeholder"
-              type="text"
+              type="number"
               id="rating"
-              name="rating"
+              step={.1}
+              {...register('rating',
+                {
+                  required: { value: true, message: 'Rating is required' },
+                  max: { value: 10, message: 'Allowed value is 0-10' },
+                  min: { value: 0, message: 'Allowed value is 0-10' }
+                })
+              }
               placeholder="Enter the movie rating"
-              value={formData.rating}
-              onChange={handleInputChange}
             />
+            <span className='form-error-message'>{errors?.rating?.message}</span>
           </div>
           <div className="form-field">
             <label className="form-label" htmlFor="genre">
               GENRE
             </label>
-            <select
-              className="form-select"
-              id="genre"
-              name="genre"
-              value={formData.genre}
-              onChange={handleInputChange}
-            >
-              <option value="action">Action</option>
-              <option value="comedy">Comedy</option>
-              <option value="drama">Drama</option>
-              <option value="sci-fi">Sci-Fi</option>
-            </select>
+            <Select
+              classNamePrefix='form-select'
+              {...register('genres', { required: true })} // Use register properly
+              options={[
+                { value: 'Action', label: 'Action' },
+                { value: 'Comedy', label: 'Comedy' },
+                { value: 'Drama', label: 'Drama' },
+                { value: 'Horror', label: 'Horror' },
+                { value: 'Science Fiction', label: 'Science Fiction' },
+              ]}
+              isMulti
+              value={selectedGenre} // Set the value prop to avoid the mentioned error
+              onChange={(selectedOptions) => setValue('genres', selectedOptions)} // Update the genre value on change
+            />
+            {
+              (!selectedGenre || selectedGenre.length) <1 && 
+              <span className='form-error-message'>Select at lease one genre to proceed</span>
+            }
           </div>
           <div className="form-field">
             <label className="form-label" htmlFor="runtime">
               RUNTIME
             </label>
             <input
               className="form-input form-placeholder"
-              type="text"
+              type="number"
               id="runtime"
-              name="runtime"
+              step={1}
+              {...register('runtime', { required: 'Runtime is required' },
+                { min: 1, message: 'Allowed value is 0-1000' },
+                { max: 1000, message: 'Allowed value is 0-1000' })
+              }
```

Comment:
Write this code in a separate method and call it over here.

Reply:
Created separate method.